### PR TITLE
Feat: Add server to get redis cluster info

### PR DESCRIPTION
**Pull request by DinaYakovlev**
Adding two server endpoints:
`GET /info` - return the view of the cluster, nodes and followers
image

`Get /state` - return the state of the cluster
image

### CODE REVIEW

**Comment by voltbit** [comment]
Thanks for working on this!
Could you please add to the description of the PR examples of the output of those two endpoints?

File: Dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ WORKDIR /workspace
 # install curl
 RUN apt-get update \
     && apt-get install -y curl
-
+
```
Comment (voltbit) [inline Dockerfile:11]: Some leftover whitespace here.

File: hack/install [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+kind --name $CLUSTER_NAME get kubeconfig > "$CLUSTER_NAME.kubeconfig.yaml"
+
+current_context=$(kubectl config current-context)
+if [ "$current_context" = "kind-$CLUSTER_NAME" ]; then
+  # add metrics server to the cluster
+  kubectl apply -f addons/metrics-server.yaml
+  kubectl patch deployment metrics-server -n kube-system -p '{"spec":{"template":{"spec":{"containers":[{"name":"metrics-server","args":["--cert-dir=/tmp", "--secure-port=4443", "--kubelet-insecure-tls","--kubelet-preferred-address-types=InternalIP"]}]}}}}'
+else
+  echo "Please set the current cluster context to kind-$CLUSTER_NAME and re-run the install script"
+fi
```
Comment (voltbit) [inline hack/install:14]: This file is probably not needed.

File: main.go [hunk 3]
```diff
@@ -39,6 +40,12 @@ func getRedisCLI(log *logr.Logger) *rediscli.RedisCLI {
 }
 
 func main() {
+	go startManager()
```
Comment (voltbit) [inline main.go:43]: I guess it doesnt matter too much which of the paths are spawned in a goroutine and which in the main. 

However, I would personally see it more fitting to have the manager starting in main and the server start on a goroutine. I see the server more as a background process.

File: config/manager/base/manager.yaml [hunk 4]
```diff
@@ -36,6 +36,10 @@ spec:
           value: "admin"
         - name: REDISCLI_AUTH
           value: "adminpass"
+        - name: CLUSTER_VIEW_FILE
+          value: "cluster_data.txt"
+        - name: CLUSTER_STATE_FILE
+          value: "cluster_state.txt"
```
Comment (voltbit) [inline config/manager/base/manager.yaml:42]: If the format of the saved data is JSON maybe we can use .json extention.

**Review by voltbit — COMMENTED** [review]
looks good :+1:

File: config/manager/base/manager.yaml (hunk 4 above)
Comment (DinaYakovlev) [reply config/manager/base/manager.yaml:42]: it bytes actually, so its not really matter

File: Dockerfile (hunk 1 above)
Comment (DinaYakovlev) [reply Dockerfile:11]: deleted :)

File: main.go (hunk 3 above)
Comment (DinaYakovlev) [reply main.go:43]: yes I agree

**Review by voltbit — APPROVED** [review]

**Pull request merged**